src/main/java/org/openstreetmap/atlas/geography/atlas/geojson/LineDelimitedGeoJsonConverter.java

```diff
@@ -0,0 +1,222 @@
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.openstreetmap.atlas.geography.atlas.Atlas;
+import org.openstreetmap.atlas.geography.atlas.AtlasResourceLoader;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasEntity;
+import org.openstreetmap.atlas.geography.atlas.items.ItemType;
+import org.openstreetmap.atlas.streaming.resource.File;
+import org.openstreetmap.atlas.streaming.resource.FileSuffix;
+import org.openstreetmap.atlas.tags.RelationTypeTag;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.runtime.Command;
+import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+import org.openstreetmap.atlas.utilities.time.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+
+/**
+ * This CLI takes a directory of atlas files and turns them into line-delimited GeoJSON.
+ *
+ * @author hallahan
+ */
+public class LineDelimitedGeoJsonConverter extends Command
+{
+    // Works great on a MacBook Pro (Retina, 15-inch, Mid 2015)
+    private static final int DEFAULT_THREADS = 8;
+
+    /**
+     * After all of your files are converted to LD GeoJSON, it is then concatenated into
+     * EVERYTHING.geojson
+     */
+    private static final String EVERYTHING = "EVERYTHING.geojson";
+
+    private static final Logger logger = LoggerFactory
+            .getLogger(LineDelimitedGeoJsonConverter.class);
+
+    private static final AtlasResourceLoader ATLAS_RESOURCE_LOADER = new AtlasResourceLoader();
+
+    private static final Switch<Path> ATLAS_DIRECTORY = new Switch<>("atlasDirectory",
+            "The directory of atlases to convert.", Paths::get, Optionality.REQUIRED);
+
+    protected static final Switch<Path> GEOJSON_DIRECTORY = new Switch<>("geojsonDirectory",
+            "The directory to write line-delimited GeoJSON.", Paths::get, Optionality.REQUIRED);
+
+    protected static final Switch<Boolean> OVERWRITE = new Switch<>("overwrite",
+            "Choose to automatically overwrite a GeoJSON file if it exists at the given path.",
+            Boolean::new, Optionality.OPTIONAL, "false");
+
+    private static final Switch<Integer> THREADS = new Switch<>("threads",
+            "The number of threads to work on processing atlas shards.", Integer::valueOf,
+            Optionality.OPTIONAL, String.valueOf(DEFAULT_THREADS));
+
+    /**
+     * We only want positive edges, because the negative edge can be derived at the application
+     * level, and this encodes extraneous data that can be easily derived by the map viewer. For
+     * relations, we only want multipolygon relations, as the rest can be derived from their
+     * members.
+     */
+    private static final Predicate<AtlasEntity> PREDICATE = atlasEntity ->
+    {
+        // We only want positive atlas entities. No negative ids.
+        if (atlasEntity.getIdentifier() < 0)
```

Comment:
I suggest relying on functionality such as check for type edge and use `isMasterEdgeIdentifier` instead of < 0 check

Reply:
I looked at this, and the expression would be several added function calls, because, i'd have to first check that it's edge type, than cast it to edge, then call that function, which really, is just looking at if the id is < 0...

Let me see if it has any impact on performance...

Reply:
done